src/LightPanels/LightPanels.js

```diff
@@ -530,17 +485,16 @@ module.exports = kind(
 	*/
 	replaceAt: function (start, count, info) {
 		var panels = this.getPanels(),
-			insertBefore, commonInfo, end, idx;
+			queue = this.popQueue,
+			insertBefore, commonInfo, end;
 
 		start = start < 0 ? panels.length + start : start;
 		end = start + count;
 		insertBefore = panels[end];
 		commonInfo = {addBefore: insertBefore};
+		queue = queue || [];
 
-		// remove existing panels
-		for (idx = end - 1; idx >= start; idx--) {
-			this.removePanel(panels[idx]);
-		}
+		queue = queue.concat(panels.splice(start, end - start));
```

Comment:
Also, is it OK that we don't call removePanel() anymore?  This won't call panel.destroy(), for one thing.